### PR TITLE
Kotlin 1.3.20 and fix configuration check exception

### LINKED ISSUE #801: OkBuck throws exception with Kotlin 1.3.20
```
Caused by: java.lang.IllegalStateException: No valid configuration found for 'kapt' in project 'project ':tooling:intellij:intellij-patcher''
        at com.uber.okbuck.core.annotation.AnnotationProcessorCache.getConfiguration(AnnotationProcessorCache.java:183)
        at com.uber.okbuck.core.annotation.AnnotationProcessorCache.hasEmptyAnnotationProcessors(AnnotationProcessorCache.java:117)
        at com.uber.okbuck.core.model.jvm.JvmTarget.getAptScopeForConfiguration(JvmTarget.java:129)
        at com.uber.okbuck.core.model.jvm.JvmTarget.getApt(JvmTarget.java:117)
        at com.uber.okbuck.composer.jvm.JvmLibraryRuleComposer.compose(JvmLibraryRuleComposer.java:29)
        at com.uber.okbuck.generator.BuckFileGenerator.createRules(BuckFileGenerator.java:122)
        at com.uber.okbuck.generator.BuckFileGenerator.lambda$createRules$1(BuckFileGenerator.java:81)
        at com.google.common.collect.SingletonImmutableBiMap.forEach(SingletonImmutableBiMap.java:65)
        at com.uber.okbuck.generator.BuckFileGenerator.createRules(BuckFileGenerator.java:73)
        at com.uber.okbuck.generator.BuckFileGenerator.generate(BuckFileGenerator.java:50)
        at com.uber.okbuck.OkBuckGradlePlugin.lambda$apply$7(OkBuckGradlePlugin.java:310)
```

The said project doesn't use kapt or annotationProcessor

```gradle
apply plugin: "java-library"
apply plugin: "org.jetbrains.kotlin.jvm"
apply plugin: "application"

dependencies {
    implementation deps.build.commonsCodec
    implementation deps.build.commonsIo
    implementation deps.build.javassist
    implementation deps.build.ztZip
    implementation deps.kotlin.stdLib
    implementation deps.kotlin.stdLibJdk8

    testImplementation project(":libraries:foundation:test:test-core")
}

mainClassName = "com.ubercab.intellijpatcher.IntellijPatcher"

run {
    args System.getProperty("exec.args", "").split(",")
}
```

### PR DESCRIPTION
**Pull request by ZacSweers**
Resolves #801

Kapt can be absent, and existing code seems to think this is possible as well, so we shouldn't throw an exception here and just gracefully handle it

Also make changes to how we consume `kotlin-compiler-embeddable` artifact. From version 1.3.20 onwards, it has a runtime dependency on trove4j which is not bundled into the jar due to licensing constraints. Fetch the runtime dependencies separately and create a new `kotlin-compiler-embeddable` fat jar with all required runtime dependencies while setting up the kotlin home in okbuck

### CODE REVIEW

File: buildSrc/src/main/java/com/uber/okbuck/core/annotation/AnnotationProcessorCache.java [hunk 1]
```diff
@@ -114,8 +115,8 @@ public AnnotationProcessorCache(
    */
   @SuppressWarnings("BooleanMethodIsAlwaysInverted")
   public boolean hasEmptyAnnotationProcessors(Project project, String configurationString) {
-    Configuration configuration = getConfiguration(project, configurationString);
-    return hasEmptyAnnotationProcessors(project, configuration);
+    Optional<Configuration> configuration = getConfiguration(project, configurationString);
+    return configuration.map(files -> hasEmptyAnnotationProcessors(project, files)).orElse(true);
```
Comment (raviagarwal7) [inline buildSrc/src/main/java/com/uber/okbuck/core/annotation/AnnotationProcessorCache.java]:
```suggestion
    return configuration.map(config -> hasEmptyAnnotationProcessors(project, config)).orElse(false);
```
should be false if configuration is not present

**Comment by kageiit** [comment]
@ZacSweers can you please rebase?

**Comment by ZacSweers** [comment]
Rebased, but the trove dependency is still there. According to @raviagarwal7 okbuck/buck need to switch to the compiler release zip https://github.com/JetBrains/kotlin/releases/download/v1.3.20/kotlin-compiler-1.3.20.zip

This currently blocks gradle 5.1, AGP 3.3.0, and kotlin 1.3.20

**Comment by kageiit** [comment]
Seems like kotlin friend paths are not working with kotlin 1.3.20 . I have all the changes with this pr and removed the kotlin bump and friend paths worked fine which leads me to believe it is an issue with 1.3.20 not picking these up correctly.

I could not try 1.3.21 as that has a completely different issue with the classloader not being able to load classes after commit https://github.com/JetBrains/kotlin/commit/b6aecf393347b177e91be506475cc1324132ca00 (which only exists in 1.3.21 but not 1.3.20)

**Pull request merged**